### ISSUE TITLE
Design Intent - Accessibility Feedback - Accredited Representative Facing, Accredited Representative Portal, Form 21-22 approval denial

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#77353]
## Next Steps for the VFS team

- [x] **Assign** this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [x] **Comment** on this ticket:
  - If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members`.
- [x] **Close the ticket** when all feedback has been addressed.

## Thoughts/questions

- I mentioned this in the meeting, but I'll mention it again here --- the Blinded Veterans Association is one of the VSOs that will be using this tool, and I definitely want to make sure we get this right for them!

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform Governance reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- [x] **Must:** Start thinking about mobile and high-zoom displays. Even though VSOs are the anticipated audience, you won't be able to assume all users are sitting down at a full-resolution desktop. Many blind and low vision users use a tablet or smartphone as their primary (and sometimes only) device, since these often have a good screen reader and gesture control options that are quicker to learn than a braille keyboard. Likewise, many low vision users who are on desktop will have their browser zoomed in (WCAG guidance requires that everything works up to 400% zoom).
- [x] **Must:** Make sure everything has an accessible name --- meaning, every link or button or other control has some kind of text equivalent that's exposed to screen reader users. I'm especially thinking about things like the notification indicator, the settings gear, etc in the wireframes.
   
   The shortest path to providing an accessible name is just text visible on the screen alongside any iconography. Besides helping screen reader users, that removes any ambiguity about what the name of the element is ("Click Settings" versus "Click the little gear thing") and helps voice command software users interacting with the page. In situations where visible text isn't an option for design reasons, an `aria-label` that provides an accessible name just to screen reader users is okay too --- but wherever possible, aim for visible text.
- [x] **Should:** Now is a great time to start thinking about the structure and sequence of information on the page. 
   
   Sighted users typically experience the page in two dimensions -- up/down, left/right. Your eyes may wander all over the screen looking for the next thing you need, rather than following a straight-line path.
   
   Screen reader users may experience the page in a more linear way. Information comes in a sequence: something, then something else, then another thing. There are navigational techniques that work to jump around the page, but presenting information in the order in which a user is likely to need it makes life a lot easier.
   
   A good exercise: Make an outline! Without any visual elements, just make an outline of all the content you expect to have on the page and work through a task you expect a VSO to complete. If they can complete the task without ever having to go backwards up the outline, that's great.
   
   Your outline is also going to inform the heading structure of the page. Many screen reader users navigate a page by having their screen reader announce all of the headings on the page and then skipping to the heading that describes the section they're looking for. Each page should have one H1 that describes the page, then H2s to describe each big section of the page, then H3s that describe subsections of each big section, and so on. Note that headings can be visually hidden using the `sr-only` class. As I said above, the shortest path to accessibility is visible text. But where necessary, heading structure can be provided to screen reader users when other visual elements provide information about structure to sighted users.
- [x] **Should:** Whenever you've got a more complex interaction, feel free to post in the #accessibility-help channel in Slack or reach out to me! From your wireframes, I'm thinking about the Permissions page [Admin view] with the table-with-checkboxes layout. That's not a standard pattern in use on VA.gov and it can be a little tricky to use with assistive technology. Definitely run any ideas by an accessibility specialist to look for potential problem spots.

## Governance team actions
- [x] Format feedback as individual tasks (check boxes)
- [x] Assign this ticket to the VFS team member that opened the Slack request
- [x] Add the VFS team product label
- [x] Add the VFS team the feature label (if applicable)
- [x] Add the touchpoint labels
- [x] Add the practice area labels
- [x] Add the Collaboration Cycle initiative milestone
- [x] Add the Collaboration Cycle initiative epic

**Comment by kristen101606** [comment]
Thanks for these notes. Our responses are below: 

From CAIA:
> Must: Start thinking about mobile and high-zoom displays. Even though VSOs are the anticipated audience, you won't be able to assume all users are sitting down at a full-resolution desktop. Many blind and low vision users use a tablet or smartphone as their primary (and sometimes only) device, since these often have a good screen reader and gesture control options that are quicker to learn than a braille keyboard. Likewise, many low vision users who are on desktop will have their browser zoomed in (WCAG guidance requires that everything works up to 400% zoom).
- **Our response:** Thanks for flagging this. As we moved from the initial wireframes we showed at the Design Intent meeting to [higher fidelity mockups](https://www.figma.com/design/LVCQBuW7a6nfVFNyhA4kv4/ARF---Design-Explorations?node-id=68-94780&t=G0A1f91ciGd3a0pd-1), we tried to ensure that related content was grouped together so that it would stay in close proximity, even when magnified. We're also using VADS components, where possible, which we think will scale properly with magnification. [Will, are you aware of any other actions we should take to account for magnification?] If you see any specific areas that we need to reconsider based on usability with magnification, please let us know.

2. From CAIA:
> Must: Make sure everything has an accessible name --- meaning, every link or button or other control has some kind of text equivalent that's exposed to screen reader users. I'm especially thinking about things like the notification indicator, the settings gear, etc in the wireframes.
> 
> The shortest path to providing an accessible name is just text visible on the screen alongside any iconography. Besides helping screen reader users, that removes any ambiguity about what the name of the element is ("Click Settings" versus "Click the little gear thing") and helps voice command software users interacting with the page. In situations where visible text isn't an option for design reasons, an aria-label that provides an accessible name just to screen reader users is okay too --- but wherever possible, aim for visible text.
- **Our response:** As we've increased the fidelity of the mockups, we've added labels to elements that were previously just images. In cases where we only have an icon (like with the login icon on mobile screens), we've noted that we'll need to double check for clear alt text.

3. From CAIA:
> Now is a great time to start thinking about the structure and sequence of information on the page.
> 
> Sighted users typically experience the page in two dimensions -- up/down, left/right. Your eyes may wander all over the screen looking for the next thing you need, rather than following a straight-line path.
> 
> Screen reader users may experience the page in a more linear way. Information comes in a sequence: something, then something else, then another thing. There are navigational techniques that work to jump around the page, but presenting information in the order in which a user is likely to need it makes life a lot easier.
> 
> A good exercise: Make an outline! Without any visual elements, just make an outline of all the content you expect to have on the page and work through a task you expect a VSO to complete. If they can complete the task without ever having to go backwards up the outline, that's great.
> 
> Your outline is also going to inform the heading structure of the page. Many screen reader users navigate a page by having their screen reader announce all of the headings on the page and then skipping to the heading that describes the section they're looking for. Each page should have one H1 that describes the page, then H2s to describe each big section of the page, then H3s that describe subsections of each big section, and so on. Note that headings can be visually hidden using the sr-only class. As I said above, the shortest path to accessibility is visible text. But where necessary, heading structure can be provided to screen reader users when other visual elements provide information about structure to sighted users.
- **Our response:** Thanks for the notes. We've annotated our [mockups](https://www.figma.com/design/LVCQBuW7a6nfVFNyhA4kv4/ARF---Design-Explorations?node-id=68-94780&t=G0A1f91ciGd3a0pd-1) to indicate the structure of each page, including headers and in one ambiguous case, reading order. Please let us know at Midpoint Review if any of the annotations are confusing or counter to best-practice.

4. From CAIA:
> Should: Whenever you've got a more complex interaction, feel free to post in the #accessibility-help channel in Slack or reach out to me! From your wireframes, I'm thinking about the Permissions page [Admin view] with the table-with-checkboxes layout. That's not a standard pattern in use on VA.gov and it can be a little tricky to use with assistive technology. Definitely run any ideas by an accessibility specialist to look for potential problem spots.
- **Our response:** Thanks! After much deliberation, we moved away from a table view and used cards to present the data instead. We appreciate having access to the #accessibility-help channel and will use this as questions come up!
